src/test/java/com/ui_init_setup/practiceproject/pages/SwagLabsLoginPage.java

```diff
@@ -53,8 +58,22 @@ private void setPassword(String password) {
         inputPassField.sendKeys(password);
     }
 
+    public String getErrorMessage() {
+        wdWait.until(ExpectedConditions.visibilityOfElementLocated(By.className("error-message-container")));
+        return errorMessage.getText();
+    }
+
+    public boolean errorMessIsShown() {
+        return errorMessage.isDisplayed();
+    }
+
     public SwagLabsInventoryPage loginWithCredentials(String username, String password) {
         loginAsValidUser(username, password);
         return new SwagLabsInventoryPage(driver);
     }
+
+    public SwagLabsLoginPage loginWithNotValidCredentials(String username, String password) {
```

Comment:
Here we seem to have contradiction in naming: a "loginWithNotValidCredentials" method that calls a "loginAsValidUser", while passing invalid credentials. Perhaps simply rename the "loginAsValidUser" to "logInAs"?